### PR TITLE
Add frost-bookends component

### PR DESCRIPTION
**Pull request by job13er**
**This project uses [semver](http://semver.org), please check the scope of this pr:**

- [ ] #none# - documentation fixes and/or test additions
- [ ] #patch# - backwards-compatible bug fix
- [x] #minor# - adding functionality in a backwards-compatible manner
- [ ] #major# - incompatible API change

# CHANGELOG

* **Added** `frost-bookends` component, implementing the pattern used by `ember-frost-modal` for a fixed header/footer with scrolling content in between. Complete with the shadow effect on the header/footer when content is scrolled beneath it. One notable difference between the styling of `frost-bookends` and that of `ember-frost-modal` is the lack of transparency in the footer, with content scrolling beneath it. This was intentional, as directed by UX (@baconC)
* **Refactored** tests for `frost-password` to make them a little DRYer, and added a test (currently only in Chrome) for the feature that maintains selected text across reveal toggles.

### CODE REVIEW

File: addon/templates/components/frost-bookends.hbs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  onScrollDown=(action 'showContentBeneathHeaderEffect')
+  onScrollUp=(action 'showContentBeneathFooterEffect')
+  onScrollYEnd=(action 'hideContentBeneathFooterEffect')
+  onScrollYStart=(action 'hideContentBeneathHeaderEffect')
+}}
+  <div
+    class='{{concat css '-content'}}'
+    data-test={{hook (concat hookPrefix '-contentWrapper') hookQualifiers}}
+  >
+    {{component content
```
Comment (sglanzer-fire) [inline addon/templates/components/frost-bookends.hbs]: Would you mind throwing a
```
{{#if hasBlock}}
  {{yield}}
{{else}}
  {{component content...
```

Here to support cases where the content doesn't do well with a component helper?

File: docs/frost-bookends.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
+# frost-bookends <br />
+
+ * [API](#api)
+ * [Examples](#examples)
+
+## API
+
+| Attribute   | Type | Value | Description |
+| ----------- | ---- | ----- | ----------- |
+| `content` | `component` | | **required** - component to render as the content (between header and footer) |
```
Comment (sglanzer-fire) [inline docs/frost-bookends.md]: As noted above around the yield case, I think we could make the `content` optional to support blockless/block formats

**Comment by sglanzer-fire** [comment]
Looks good, just the suggestion to add a `block` option as well, but that's not required, we could add it later

**Comment by sglanzer-fire** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/443/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)

File: addon/templates/components/frost-bookends.hbs (hunk 1 above)
Comment (job13er) [reply addon/templates/components/frost-bookends.hbs]: Hmm. I was actually running into that myself when consuming this. Was gonna go down the route of a computed property for the content, but I don't mind that at all.

File: docs/frost-bookends.md (hunk 2 above)
Comment (job13er) [reply docs/frost-bookends.md]: Sounds good. I was thinking we couldn't do block b/c of the need for header/footer, but I guess it makes sense to have the content be the inner block stuff.

**Comment by job13er** [comment]
@sglanzer Added that feature, please take a look when you have a minute

**Comment by job13er** [comment]
Approved

[Approved with PullApprove](https://pullapprove.com/ciena-frost/ember-frost-core/pull-request/443/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=ciena-frost/ember-frost-core)

**Pull request merged**